src/Eagle.ts

```diff
@@ -714,30 +699,48 @@ export class Eagle {
         if( selection instanceof Edge){
             GraphRenderer.setPortPeekForEdge(selection,!alreadySelected)
         }
+    }
 
-        //special case if we are selecting multiple things in a palette
-        if(selectedLocation === Eagle.FileType.Palette){
-            this.hierarchyMode(false)
-            this.rightWindow().mode(Eagle.RightWindowMode.Inspector) 
-            return
-        }
+    getObjectInspectorVisibility : ko.PureComputed<boolean> = ko.pureComputed(() => {
+        const eagle = Eagle.getInstance()
+        return eagle.selectedNode() !== null || eagle.selectedEdge() !== null || eagle.selectedObjects().length > 1
+    }, this);
+
+    getInspectorCollapseBtnState : ko.PureComputed<boolean> = ko.pureComputed(() => {
+        $('#inspector').addClass('inspectorTransition')
+        setTimeout(function(){
+            $('#inspector').removeClass('inspectorTransition')
+        },100)
 
-        if(rightWindowMode === Eagle.RightWindowMode.Hierarchy){
-            this.hierarchyMode(true)
+        if(this.getObjectInspectorVisibility()){
+            if(Setting.findValue(Setting.OBJECT_INSPECTOR_COLLAPSED_STATE)){
```

Comment:
These two ifs can each be simplified to:

return Setting.findValue(Setting.OBJECT_INSPECTOR_COLLAPSED_STATE)

and

return Setting.findValue(Setting.GRAPH_INSPECTOR_COLLAPSED_STATE)

or if the settings are of type string, then you can do:

return Setting.findValue(Setting.GRAPH_INSPECTOR_COLLAPSED_STATE) === "true"